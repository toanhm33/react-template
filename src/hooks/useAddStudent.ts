import { useMutation, useQueryClient } from "react-query";
import studentApi from "apis/service/student";

// export const useAddStudents = () => {
    // import { useMutation, useQueryClient } from 'react-query';
    // import { postBook } from 'src/apis/service/book';
    export const useAddStudents = () => {
      const queryClient = useQueryClient();
      return useMutation(studentApi.create, {
        // When mutate is called:
        onMutate: async (newBook) => {
          // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
          await queryClient.cancelQueries('todo');
    
          // Snapshot the previous value
          const previousBooks = queryClient.getQueryData<any[]>('todo');
    
          //   // Optimistically update to the new value
          if (previousBooks) {
            queryClient.setQueryData<any[]>('todo', () => [
              ...previousBooks,
              newBook,
            ]);
          }
    
          return { previousBooks };
        },
        // If the mutation fails, use the context returned from onMutate to roll back
        onError: (err, variables, context: any) => {
          if (context?.previousBooks) {
            queryClient.setQueryData<any>('todo', context.previousBooks);
          }
        },
        // Always refetch after error or success:
        onSettled: () => {
          queryClient.invalidateQueries('todo');
        },
      });
    };
    