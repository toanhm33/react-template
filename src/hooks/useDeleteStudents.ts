import studentApi from "apis/service/student";
import { useMutation, useQueryClient } from "react-query";

    export const useDeleteStudents = () => {
      const queryClient = useQueryClient();
      return useMutation((book) => studentApi.deleteById(book), {
        // When mutate is called:
        onMutate: async (booksDelete: number[]) => {
          // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
          await queryClient.cancelQueries('todo');
    
          // Snapshot the previous value
          const previousBooks = queryClient.getQueryData<any[]>('todo');
    
          //   // Optimistically update to the new value
          if (previousBooks) {
            queryClient.setQueryData<any[]>('todo', () => {
              return previousBooks.filter((book) => !booksDelete.includes(book.id));
            });
          }
          console.log(previousBooks, 'bookUpdate');
          return { previousBooks };
        },
        // If the mutation fails, use the context returned from onMutate to roll back
        onError: (err, variables, context: any) => {
          if (context?.previousBooks) {
            queryClient.setQueryData<any>('todo', context.previousBooks);
          }
        },
        // Always refetch after error or success:
        onSettled: () => {
          queryClient.invalidateQueries('todo');
        },
      });
    };
    