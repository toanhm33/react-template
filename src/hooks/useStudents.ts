import studentApi from "api/studentApi";
import { Student } from "models";
import { useQuery } from "react-query";

export const useStudents = (getPageNumber: number, onSuccess: Function, onError: Function) => {
    return useQuery<Student[], Error>(
        "query-tutorials",
        () => {
            return studentApi.findAll({
            _page: getPageNumber,
            _limit: 10,
            });
        },
        {
            enabled: false,
            retry: false,
            onSuccess: (res) => {
                onSuccess(res);
            },
            onError: (err: any) => {
                onError();
            },
        }
    );
}