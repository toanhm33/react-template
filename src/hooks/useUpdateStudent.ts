import { useMutation, useQueryClient } from "react-query";
import studentApi from "apis/service/student";

// export const useUpdateStudents = () => {
// import { useMutation, useQueryClient } from 'react-query';
// import { updateBook } from 'src/apis/service/book';

    export const useUpdateStudents = () => {
      const queryClient = useQueryClient();
      return useMutation((book) => studentApi.update(book), {
        // When mutate is called:
        onMutate: async (bookUpdate: any) => {
          // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
          await queryClient.cancelQueries('todo');
    
          // Snapshot the previous value
          const previousBooks = queryClient.getQueryData<any[]>('todo');
    
          //   // Optimistically update to the new value
          if (previousBooks) {
            queryClient.setQueryData<any[]>('todo', () => {
              const previousBooksCopy = [...previousBooks];
              const foundBook = previousBooksCopy.findIndex(
                (book) => book.id === bookUpdate.id
              );
              if (foundBook) {
                previousBooksCopy[foundBook] = bookUpdate;
              }
              return previousBooksCopy;
            });
          }
    
          return { previousBooks };
        },
        // If the mutation fails, use the context returned from onMutate to roll back
        onError: (err, variables, context: any) => {
          if (context?.previousBooks) {
            queryClient.setQueryData<any>('todo', context.previousBooks);
          }
        },
        // Always refetch after error or success:
        onSettled: () => {
          queryClient.invalidateQueries('todo');
        },
      });
    };
    