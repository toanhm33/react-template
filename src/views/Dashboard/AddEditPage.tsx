import studentApi from 'api/studentApi';
import { Student } from 'models';
import * as React from 'react';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import StudentForm from 'views/component/StudentForm';

export  interface AddEditPageProps {
}

function AddEditPage (props: AddEditPageProps) {
  const { student_id } = useParams<{ student_id: string }>();
  const [student, setStudent] = useState<Student>();
  const params = useParams();

  const isEdit = Boolean(student_id);

  useEffect(() => {
    if(!student_id) return;
    
    (async () => {
      try {
        const data: Student = await studentApi.getById(student_id);
      } catch (error) {}})();
  },[student_id]);

  const inititalValues: Student = {
    name: '',
    age: '',
    mark: '',
    gender: '',
    city: '',
    ...student,
  } as Student;
  const handleStudentFormSubmit = (formValues: Student) => {};

  return (
      <div>
        <h1>{isEdit ? 'Update student' : 'Add new studentent'}</h1>
        <div>
          <StudentForm inititalValues={inititalValues} onSubmit={handleStudentFormSubmit} />
        </div>
      </div>
  );
}

export default AddEditPage