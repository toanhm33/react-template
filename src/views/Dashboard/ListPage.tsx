import Table from 'components/Common/Table';
import * as React from 'react';
import { useQuery } from 'react-query';
import studentApi from 'api/studentApi';
import { Student } from 'models';
import { useEffect, useState } from 'react';
import Pagination from 'components/Common/Pagination';
import { Link, useHistory, useRouteMatch } from 'react-router-dom';

export interface ListPageProps {
}

export function ListPage (props: ListPageProps) {
  const [getResult, setGetResult] = useState<string | any>(null);
  const [getPageNumber, setGetPageNumber] = useState<number>(1);
  const [currentPage, setCurrentPage] = useState(1);
  const [getTitle, setGetTitle] = useState("");
  const [postsPerPage] = useState(10);
  const match = useRouteMatch();
  const history = useHistory();

  const fortmatResponse = (res: any) => {
    return JSON.stringify(res, null, 2);
  };

  const { isLoading: isLoadingTutorials, refetch: getAllTutorials } = useQuery<Student[], Error>(
    "query-tutorials",
    async () => {
      return await studentApi.findAll({
        _page: getPageNumber,
        _limit: 10,
      });
    },
    {
      enabled: false,
      onSuccess: (res) => {
        setGetResult(res);
      },
      onError: (err: any) => {
        setGetResult(fortmatResponse(err.response?.data || err));
      },
    }
  );

  useEffect(() => {
    getAllData();
  }, [isLoadingTutorials, getPageNumber]);
  
  function getAllData() {
    try {
      getAllTutorials();
    } catch (err) {
      setGetResult(fortmatResponse(err));
    }
  }

  const { isLoading: isSearchingTutorial, refetch: findTutorialsByTitle } = useQuery<Student[], Error>(
    "query-tutorials-by-title", // ["query-tutorials-by-title", getTitle],
    async () => {
      return await studentApi.findByName(getTitle);
    },
    {
      enabled: false,
      retry: 1,
      onSuccess: (res) => {
        setGetResult(res);
      },
      onError: (err: any) => {
        setGetResult(fortmatResponse(err.response?.data || err));
      },
    }
  );

  useEffect(() => {
    if (isSearchingTutorial) setGetResult("searching...");
  }, [isSearchingTutorial]);

  function getDataByTitle() {
    if (getTitle) {
      try {
        findTutorialsByTitle();
      } catch (err) {
        setGetResult(fortmatResponse(err));
      }
    }
  }

  const handleAddStudent = () => {
    history.push('/student/add');
  }

  const handleEditStudent = (student: Student) => {    
    history.push(`/student/${student.id}`);
  }

  const paginate = (pageNumber: number) => {
    setGetPageNumber(pageNumber);
  }
  // const { data, isFetching, isLoading, error, isError } = useQuery('key_unique', getPosts)
  return (
    <div>
      {isLoadingTutorials ? 
        <div
          className="spinner"
        ></div> : ''
      }
      <div className="mt-12 mb-4 flex justify-between	">
        <div className="relative inline-block text-gray-600">
          <input 
            type="search" 
            value={getTitle}
            onChange={(e) => setGetTitle(e.target.value)}
            name="search" placeholder="Search..." className="bg-white h-10 px-5 pr-10 rounded-full text-sm focus:outline-none"/>
          <button type="submit" className="absolute right-0 top-0 mt-3 mr-4">
            <svg xmlns="http://www.w3.org/2000/svg" className=" h-4 w-4" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" />
            </svg>
          </button>
        </div>
        <button onClick={handleAddStudent} className="btn btn-blue ">Add new student</button>
      </div>
      <Table 
      onEdit={handleEditStudent}
      listStudent={getResult} />
      <Pagination       
      totalPosts={getResult !== null ? getResult?.pagination?._totalRows : 0}
      postsPerPage={postsPerPage}
      paginate={paginate}
      currentPage={currentPage}
      />
    </div>
  );
}

export default ListPage