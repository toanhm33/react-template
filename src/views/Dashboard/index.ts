export * from './ListPage'
export * from './AddEditPage'