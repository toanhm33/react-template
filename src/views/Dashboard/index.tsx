import Table from 'components/Common/Table';
import * as React from 'react';
import { useQuery } from 'react-query';
import studentApi from 'api/studentApi';
import { Student } from 'models';
import { useEffect, useState } from 'react';
import Pagination from 'components/Common/Pagination';
import { useStudents } from 'hooks/useStudents';

export interface DashboardProps {
}

export function Dashboard (props: DashboardProps) {
  const [getResult, setGetResult] = useState<string | any>(null);
  const [getPageNumber, setGetPageNumber] = useState<number>(1);
  const [currentPage, setCurrentPage] = useState(1);
  const [postsPerPage] = useState(10);
  const fortmatResponse = (res: any) => {
    return JSON.stringify(res, null, 2);
  };

  const { isLoading: isLoadingTutorials, refetch: getAllTutorials } = useStudents(getPageNumber, setGetResult, (err: any) => {
    fortmatResponse(err.response?.data || err)
  });

  useEffect(() => {
    getAllData();
  }, [isLoadingTutorials, getPageNumber]);
  
  function getAllData() {
    try {
      getAllTutorials();
    } catch (err) {
      setGetResult(fortmatResponse(err));
    }
  }

  // Get current posts
  const indexOfLastPost = currentPage * postsPerPage;
  const indexOfFirstPost = indexOfLastPost - postsPerPage;

  // Change page
  const paginateFront = () => setCurrentPage(currentPage + 1);
  const paginateBack = () => setCurrentPage(currentPage - 1);

  const paginate = (pageNumber: number) => {
    setGetPageNumber(pageNumber);
  }
  // const { data, isFetching, isLoading, error, isError } = useQuery('key_unique', getPosts)
  return (
    <div>
        {isLoadingTutorials ? 'show loading' : ''}
       
        <div className=" bg-gray-200">
            <div className="container flex justify-center items-center">
                <div className="relative">
                    <input type="text" className="h-14 w-96 pl-10 pr-20 rounded-lg z-0 focus:shadow focus:outline-none" placeholder="Search anything..."/>
                    <div className="absolute top-2 right-2"> <button className="h-10 w-20 text-white rounded-lg bg-red-500 hover:bg-red-600">Search</button> </div>
                </div>
            </div>
        </div>

        <Table listStudent={getResult}/>
        <Pagination       
        totalPosts={getResult !== null ? getResult?.pagination?._totalRows : 0}
        postsPerPage={postsPerPage}
        paginate={paginate}
        currentPage={currentPage}
        />
    </div>
  );
}

export default Dashboard