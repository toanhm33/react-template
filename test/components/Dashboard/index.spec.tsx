import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { useStudents } from '../../../src/hooks/useStudents';

import { Dashboard } from '../../../src/views/Dashboard';
jest.useFakeTimers()
const mockedUseProduct = useStudents as jest.Mock<any>; 

// Mock the module
jest.mock("../../../src/hooks/useStudents");

describe('Dashboard Component', () => {
  beforeEach(() => {
    mockedUseProduct.mockImplementation(() => ({ isLoading: true }));
  })
  afterEach(() => {
		jest.clearAllMocks();
	});
  it("Renders without crashing", () => {
    const queryClient = new QueryClient();
    const dummyComponentProps = {
      productId: 1
    };
		const componentRenderer = renderer.create(<QueryClientProvider client={queryClient}>
      <Dashboard {...dummyComponentProps} />
      </QueryClientProvider>);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
	});
  it("Displays loading indicator", () => {
		const { getByText } = render(<Dashboard />);
		expect(getByText(/show loading/i)).toBeTruthy();
	});

	it("Displays error message", () => {
    const error = true;
    expect(error).toBe(true);
    // TODO CHECK ERROR
	});
  it("Displays data", () => {
		const mockedProductData = [{
      "id": "sktwi1cgkkuif36f3",
      "name": "Callie Christiansen",
      "age": 27,
      "mark": 6.8,
      "gender": "male",
      "createdAt": 1633700485643,
      "updatedAt": 1636888141276,
      "city": "hcm"
    }];
		mockedUseProduct.mockImplementation(() => ({ isLoading: false, data: mockedProductData }));

		const { queryByText } = render(<Dashboard />);

		expect(queryByText(/show loading/i)).toBeFalsy();
    // TODO CHECK DATA
	});
});



// describe('Dashboard component', () => {
//   test('should render component properly', () => {
//     // given
//     const queryClient = new QueryClient();
//     const dummyComponentProps = {
//       productId: 1
//     };

//     // when
//     const componentRenderer = renderer.create(
//     <QueryClientProvider client={queryClient}>
//       <Dashboard {...dummyComponentProps} />
//       </QueryClientProvider>
//     );
//     // jest.mock('react-query', () => ({ useQuery: () => ({ isLoading: false, error: {}, data: [], }), }));ss
//     const tree = componentRenderer.toJSON();

//     // then
//     expect(tree).toMatchSnapshot();
//   });
// });