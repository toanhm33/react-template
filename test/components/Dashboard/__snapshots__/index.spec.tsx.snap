// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component Renders without crashing 1`] = `
<div>
  show loading
  <div
    className=" bg-gray-200"
  >
    <div
      className="container flex justify-center items-center"
    >
      <div
        className="relative"
      >
        <input
          className="h-14 w-96 pl-10 pr-20 rounded-lg z-0 focus:shadow focus:outline-none"
          placeholder="Search anything..."
          type="text"
        />
        <div
          className="absolute top-2 right-2"
        >
           
          <button
            className="h-10 w-20 text-white rounded-lg bg-red-500 hover:bg-red-600"
          >
            Search
          </button>
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
      >
        <div
          className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
        >
          <table
            className="min-w-full divide-y divide-gray-200"
          >
            <thead
              className="bg-gray-50"
            >
              <tr>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Name
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Title
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Status
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Role
                </th>
                <th
                  className="relative px-6 py-3"
                  scope="col"
                >
                  <span
                    className="sr-only"
                  >
                    Edit
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              className="bg-white divide-y divide-gray-200"
            >
              
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="py-2"
  >
    <div>
      <p
        className="text-sm text-gray-700"
      >
        Showing
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        to
        <span
          className="font-medium"
        >
           
          10
           
        </span>
        of
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        results
      </p>
    </div>
    <nav
      className="block"
    >
      <ul
        className="flex pl-0 rounded list-none flex-wrap"
      >
        <li />
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Dashboard component should render component properly 1`] = `
<div>
  
  <div
    className=" bg-gray-200"
  >
    <div
      className="container flex justify-center items-center"
    >
      <div
        className="relative"
      >
        <input
          className="h-14 w-96 pl-10 pr-20 rounded-lg z-0 focus:shadow focus:outline-none"
          placeholder="Search anything..."
          type="text"
        />
        <div
          className="absolute top-2 right-2"
        >
           
          <button
            className="h-10 w-20 text-white rounded-lg bg-red-500 hover:bg-red-600"
          >
            Search
          </button>
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
      >
        <div
          className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
        >
          <table
            className="min-w-full divide-y divide-gray-200"
          >
            <thead
              className="bg-gray-50"
            >
              <tr>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Name
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Title
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Status
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Role
                </th>
                <th
                  className="relative px-6 py-3"
                  scope="col"
                >
                  <span
                    className="sr-only"
                  >
                    Edit
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              className="bg-white divide-y divide-gray-200"
            >
              
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="py-2"
  >
    <div>
      <p
        className="text-sm text-gray-700"
      >
        Showing
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        to
        <span
          className="font-medium"
        >
           
          10
           
        </span>
        of
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        results
      </p>
    </div>
    <nav
      className="block"
    >
      <ul
        className="flex pl-0 rounded list-none flex-wrap"
      >
        <li />
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Dashboard demo 2 Renders without crashing 1`] = `
<div>
  show loading
  <div
    className=" bg-gray-200"
  >
    <div
      className="container flex justify-center items-center"
    >
      <div
        className="relative"
      >
        <input
          className="h-14 w-96 pl-10 pr-20 rounded-lg z-0 focus:shadow focus:outline-none"
          placeholder="Search anything..."
          type="text"
        />
        <div
          className="absolute top-2 right-2"
        >
           
          <button
            className="h-10 w-20 text-white rounded-lg bg-red-500 hover:bg-red-600"
          >
            Search
          </button>
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
      >
        <div
          className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
        >
          <table
            className="min-w-full divide-y divide-gray-200"
          >
            <thead
              className="bg-gray-50"
            >
              <tr>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Name
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Title
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Status
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Role
                </th>
                <th
                  className="relative px-6 py-3"
                  scope="col"
                >
                  <span
                    className="sr-only"
                  >
                    Edit
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              className="bg-white divide-y divide-gray-200"
            >
              
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="py-2"
  >
    <div>
      <p
        className="text-sm text-gray-700"
      >
        Showing
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        to
        <span
          className="font-medium"
        >
           
          10
           
        </span>
        of
        <span
          className="font-medium"
        >
           
          0
           
        </span>
        results
      </p>
    </div>
    <nav
      className="block"
    >
      <ul
        className="flex pl-0 rounded list-none flex-wrap"
      >
        <li />
      </ul>
    </nav>
  </div>
</div>
`;
